src/main/java/carpet/script/CarpetExpression.java

```diff
@@ -2326,15 +2339,15 @@ public String getString()
                     );
                     return (_c, _t ) -> retval;
                 }
-                else if (lv.size()==4)
+                else if (lv.size()==2+cposLocator.offset)
                 {
-                    width = (int) ((NumericValue) lv.get(3).evalValue(c)).getLong();
+                    width = (int) ((NumericValue) lv.get(lv.size()-1).evalValue(c)).getLong();
                     height = 0;
                 }
-                else // size == 5
+                else // size == 3 + cposLocator.offset
                 {
-                    width = (int) ((NumericValue) lv.get(3).evalValue(c)).getLong();
-                    height = (int) ((NumericValue) lv.get(4).evalValue(c)).getLong();
+                    width = (int) ((NumericValue) lv.get(lv.size()-2).evalValue(c)).getLong();
```

Comment:
again - don't count list from the back, just use offsets to count from the front.